src/lock.rs

```diff
@@ -62,11 +62,17 @@ pub struct Lock<'a> {
     pub lock_manager: &'a LockManager,
 }
 
+#[cfg(not(feature = "tokio-comp"))]
 #[derive(Debug, Clone)]
 pub struct LockGuard<'a> {
     pub lock: Lock<'a>,
 }
 
+
+// Dropping this guard inside the context of a tokio runtime if tokio-comp is enabled 
+// will block the tokio runtime. 
+// Because of this, the guard is not compiled if tokio-comp is enabled. 
+#[cfg(not(feature = "tokio-comp"))]
```

Comment:
```suggestion
#[cfg(feature = "async-std-comp")]
```

this prevents compilation errors when end users try installing with `no-default-features` as we required `Drop` for only `async-std-comp` and nothing else currently